src/compiler/parser.ts

```diff
@@ -2461,6 +2464,22 @@ module ts {
 
             return result;
         }
+        
+        function parseTypePredicateOrHigher(): TypeNode {
+            let type = parseUnionTypeOrHigher();
+            if (token === SyntaxKind.IsKeyword &&
+                type.kind === SyntaxKind.TypeReference &&
+                (<TypeReferenceNode>type).typeName.kind === SyntaxKind.Identifier) {

```

Comment:
You need to also check that the node has no type arguments. A `TypeReferenceNode` could have type arguments, and you don't want this to be parsed as a type predicate
```ts
Foo<T> is Bar
```